File: backend/services/auth-service/src/middleware/auth.middleware.js
import jwt from 'jsonwebtoken';
import { ApiError } from '../utils/ApiError.js';
import { ApiResponse } from '../utils/ApiResponse.js';
import { asyncHandler } from '../utils/asyncHandler.js';
import { User } from '../models/user.models.js';

import admin from '../utils/firebase.js';

// we need to rotate the token at each request to avoid token expiration

const verifyJWT = asyncHandler(async (req, res, next) => {
  const token = req.headers.authorization?.split(' ')[1];

  if (!token) {
    throw new ApiError(401, 'Unauthorized');
  }

  try {
    const decoded = await admin.auth().verifyIdToken(token);
    console.log("Decoded: ", decoded);
    if (!decoded) {
      throw new ApiError(401, 'Access Tpken Expired or invalid');
    }

    req.user = decoded;
    next();
  } catch (err) {
    throw ApiError(500, err.message);
  }
});

export { verifyJWT };

File: backend/services/ai_service/api/middleware/auth_middleware.py
# ✅ middleware/auth_middleware.py
from fastapi import Request
from jose import jwt
import os
from firebase_admin import auth
from api.utils.apiError import ApiError
from api.utils.apiResponse import ApiResponse

async def verifyJWT(request: Request):
    try:
        token = request.headers.get("Authorization").split(" ")[1]
        if not token:
            return ApiError.send(
                statusCode=401,
                data = {},
                message = "Token Not Found"
            )
            
        decoded = auth.verify_id_token(token)
        if not decoded:
            return ApiError.send(
                401,
                data={},
                message="Invalid Token"
            )
        
        print("Verify By Firebase: ",decoded)
        return decoded
    except Exception as e:
        return {"error": str(e)}
